### PR TITLE
Edit and delete - backend

### PR DESCRIPTION
**Pull request by DebbySchavl**

### CODE REVIEW

File: backend/src/main/java/com/github/debbyschavl/LocationDataController.java [hunk 1]
```diff
@@ -35,6 +39,25 @@ public Document addMongoDocs(@RequestBody String json) {
...
+    }
+
+    @PutMapping("/{id}")
+    public LocationData updateLocation(@PathVariable String id, @RequestBody Location location){
+       return locationDataService.updateLocationData(id, location);
+    }
+
+
+
+    @ExceptionHandler({NoSuchLocationDataException.class})
```
Comment (Pia0912) [inline backend/src/main/java/com/github/debbyschavl/LocationDataController.java:54]: Do you need this exception here? I think you have your own exception class "GlobalExceptionHandler" and handle those exceptions in your service class?

File: backend/src/main/java/com/github/debbyschavl/model/Location.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@Data
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class Location {
+         City city;
+         Continent continent;
+         Country country;
+
+         @Field(value = "location")
```
Comment (Pia0912) [inline backend/src/main/java/com/github/debbyschavl/model/Location.java:16]: I have this annotation too, but Florian told me you apparently don't need the @Field? I am not too sure about this one though

File: backend/src/test/java/com/github/debbyschavl/LocationDataControllerTest.java [hunk 3]
```diff
@@ -30,7 +30,7 @@ class LocationDataControllerTest {
             "\"_id" +
             "\":{" +
             "\"$oid\":\"6156dab2dbc1a403b62c5657\"}," +
-            "\"id\":\"6156dab06df2e755c3fc96d2\"," +
+            "\"iid\":\"6156dab06df2e755c3fc96d2\"," +
```
Comment (Pia0912) [inline backend/src/test/java/com/github/debbyschavl/LocationDataControllerTest.java:33]: what is this iid?

**Review by Pia0912 — APPROVED** [review]
LGTM! 🚀 If you have questions to my comments - hit me up via Slack 😆

File: backend/src/main/java/com/github/debbyschavl/model/Location.java (hunk 2 above)
Comment (DebbySchavl) [reply backend/src/main/java/com/github/debbyschavl/model/Location.java:16]: I think I really need this one: I am using external Data, where the field name "location" occurs twice, what would not work within the backend (I think), so I renamed one of the fields to GeoData. The @Filed annotation references the GeoData object in the backend to the location field in the mongoDB ...so, this is how I understand it

File: backend/src/test/java/com/github/debbyschavl/LocationDataControllerTest.java (hunk 3 above)
Comment (DebbySchavl) [reply backend/src/test/java/com/github/debbyschavl/LocationDataControllerTest.java:33]: same thing: my external data have two ids, one of them is called iid, I tried to rename it...not a good idea...so i reverted it

File: backend/src/main/java/com/github/debbyschavl/LocationDataController.java (hunk 1 above)
Comment (DebbySchavl) [reply backend/src/main/java/com/github/debbyschavl/LocationDataController.java:54]: You are right, I obviously don't need it! Thanks!

**Pull request merged**